File: commands/ping.js
let clc = require("cli-color");
let moment = require("moment");
require("moment-timezone");
require("moment-duration-format");

module.exports.run = async (client, message, args, cmdHook, roCMD, DBResult) => {
    const startMessage = await message.channel.send({embed: {
        color: 1069660,
        author: {
            name: "HolidayBot",
            icon_url: client.user.displayAvatarURL
        },
        title: "Pinging!",
        description: "Awaiting response",
        footer: {
            icon_url: message.author.displayAvatarURL,
            text: message.author.username
        }
    }});

    let messagePing = Math.round(startMessage.createdTimestamp - message.createdTimestamp);

    startMessage.edit("", {embed: {
        color: 1069660,
        author: {
            name: "HolidayBot",
            icon_url: client.user.displayAvatarURL
        },
        title: "Pong!",
        description: `API Latency ${Math.round(client.ping)}ms\nMessage Latency: ${messagePing}`,
        footer: {
            icon_url: message.author.displayAvatarURL,
            text: message.author.username
        }
    }});
    client.misc.cmdHook(message.content, "succ", null, message.author, message.guild, null);
};

module.exports.help = {
    name: "ping"
}

File: commands/Ping.go
package commands

import (
	"math"
	"strconv"

	"github.com/barkloaf/HolidayBot/misc"
	"github.com/bwmarrin/discordgo"
)

func Ping(client *discordgo.Session, interaction *discordgo.Interaction) error {
	latency := strconv.FormatFloat(math.Round(.000001*float64(client.HeartbeatLatency())), 'f', -1, 64)

	client.InteractionRespond(interaction, &discordgo.InteractionResponse{
		Type: discordgo.InteractionResponseChannelMessageWithSource,
		Data: &discordgo.InteractionResponseData{
			Embeds: []*discordgo.MessageEmbed{
				{
					Author: &discordgo.MessageEmbedAuthor{
						Name:    client.State.User.Username,
						IconURL: client.State.User.AvatarURL(""),
					},
					Color: misc.Config.UseColor,
					Title: "Pong!\n🏓  " + latency + " ms",
				},
			},
		},
	})

	return nil
}
